### PR TITLE
Deallocate resource at instance.remove

### PR DESCRIPTION
**Pull request by StrongMonkey**
To provide another option for https://github.com/rancher/cattle/pull/2427.

### CODE REVIEW

**Comment by cjellick** [comment]
@StrongMonkey You're on to something, but let's take it a step farther. Take all the logic in InstancePurge and put it at the end of InstanceRemove. Delete the instancePurge class and any references to it. Also, make PostInstancePurge PostInstanceRemove and refactor it accordingly to work as an instance.remove process. Give it a high priority like 10,000 so that it runs after the other Post handlers.

To verify that you are setting the processes up correctly, look at http://localhost:8080/v2-beta/resourcedefinitions/1rd!instance/resourcedot and assert that once you're changes are done, instance.purge has no handlers and instance.remove has the new post handler (and that handler is last).

Discalimer: @ibuildthecloud may have some reason that this wont work that I haven't thought of, but lets do it and see if we can get it to pass cattle CI.

**Comment by StrongMonkey** [comment]
@cjellick tests tweaked. Most changes are because we are moving volume.purge and link.purge into instance.remove process. By justing moving the deallocate process into instance.remove, we don't even need to change the tests.

**Comment by StrongMonkey** [comment]
@cjellick I just discussed with Darren, and Darren approves the idea of moving all the functionality from instance.purge to instance.remove. This could also address the issue https://github.com/rancher/rancher/issues/7590. 
Claim: @StrongMonkey will be responsible for anything that will break in this pr. 🙂

File: tests/integration/cattletest/core/test_volume.py [hunk 1]
```diff
@@ -334,11 +334,8 @@ def test_volume_mounting_and_delete(new_context, super_client):
     c3 = client.wait_success(c3.stop())
     c3 = client.wait_success(c3.remove())
     check_mount_count(client, c3, 0)
-    v1 = wait_for_condition(client, v1, lambda x: x.state == 'detached')
-    check_mount_count(client, v1, 0)
+    wait_for_condition(client, v1, lambda x: x.removed is not None)
```
Comment (cjellick) [inline tests/integration/cattletest/core/test_volume.py]: Flake8 error here from too many blank lines

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/7933

**Comment by StrongMonkey** [comment]
https://github.com/rancher/rancher/issues/8568

**Comment by StrongMonkey** [comment]
close in favor of https://github.com/rancher/cattle/pull/2641

**Pull request closed without merging**